packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -1184,3 +1212,82 @@ function cardsOrFieldsFromModule(
 ): (typeof BaseDef)[] {
   return Object.values(module).filter(isCardOrField);
 }
+
+function collectLocalCardsOrFields(
+  moduleSyntax: ModuleSyntax,
+  exportedCardsOrFields: (typeof BaseDef)[],
+): Map<PossibleCardOrFieldClass, typeof BaseDef> {
+  const localCardsOrFields: Map<PossibleCardOrFieldClass, typeof BaseDef> =
+    new Map();
+  let possibleCardsOrFields = moduleSyntax.possibleCardsOrFields;
+
+  for (const value of moduleSyntax.elements) {
+    const cardOrField = exportedCardsOrFields.find(
+      (c) => c.name === value.localName,
+    );
+
+    if (cardOrField !== undefined) {
+      findLocalAncestor(
+        value,
+        cardOrField,
+        possibleCardsOrFields,
+        localCardsOrFields,
+      );
+      findLocalField(
+        value,
+        cardOrField,
+        possibleCardsOrFields,
+        localCardsOrFields,
+      );
+    }
+  }
+
+  return localCardsOrFields;
+}
+
+function findLocalAncestor(
+  value: ElementDeclaration,
+  cardOrField: typeof BaseDef,
+  possibleCardsOrFields: PossibleCardOrFieldClass[],
+  localCardsOrFields: Map<PossibleCardOrFieldClass, typeof BaseDef>,
+) {
+  if (isPossibleCardOrFieldClass(value) && isInternalReference(value.super)) {
+    const indexOfParent = value.super.classIndex;
+    if (indexOfParent !== undefined) {
+      const parentCardOrFieldClass = possibleCardsOrFields[indexOfParent];
+      const parentCardOrField = getAncestor(cardOrField);
+
+      if (parentCardOrField) {
+        localCardsOrFields.set(parentCardOrFieldClass, parentCardOrField);
+      }
+    }
+  }
+}
+
+function findLocalField(
+  value: ElementDeclaration,
+  cardOrField: typeof BaseDef,
+  possibleCardsOrFields: PossibleCardOrFieldClass[],
+  localCardsOrFields: Map<PossibleCardOrFieldClass, typeof BaseDef>,
+) {
+  if (isPossibleCardOrFieldClass(value)) {
+    if (value.possibleFields) {
+      for (const [fieldName, v] of value.possibleFields) {
+        if (isInternalReference(v.card)) {
+          const indexOfParentField = v.card.classIndex;
+          if (indexOfParentField !== undefined) {
+            const parentFieldClass = possibleCardsOrFields[indexOfParentField];
+            const localName = parentFieldClass.localName;
+
+            if (localName) {
```

Comment:
Would combinations of early returns and `continue`s help taming the big nesting depth of this function?

Reply:
https://github.com/cardstack/boxel/pull/716/commits/3534856782445a647913007b0b837c178c129b2c. I didn't use continue but I did use early return